Implement Homepage v2 Redesign

vkoves opened the pull request:
# Description

Implemented the homepage design, which features more visual building tiles for the 10 highest emitting buildings and 8 featured buildings, plus a much more prominent search.

| Before | After |
| --- | --- |
| ![Screenshot from 2025-02-14 19-13-22](https://github.com/user-attachments/assets/60f11b22-2c16-4a20-a153-a15b53d4ee1c) | ![Screenshot from 2025-02-14 19-13-15](https://github.com/user-attachments/assets/23e210b8-fee4-432c-aa67-04f93ec0fc32) |

<details>
  <summary><strong>Mobile Screenshot</strong></summary>
<img src="https://github.com/user-attachments/assets/594895f8-d11d-472e-8de8-c3cb8528a604" width="300">
</details>

Fixes #155

# Testing Instructions

Tested the new homepage on desktop, mobile and a variety of screen sizes, and confirmed everything looks great and is easy to read.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

<!-- PR template modified from: https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ -->

wangbertha commented on src/pages/HighestEmissionsIntensity.vue:
This page feels a bit hidden, since it's only linked in a "View More" aside on the homepage, while the other `BuildingsTable` views are more prominently linked in the AppHeader. Maybe we can link this page in the AppHeader as well?

vkoves commented on src/components/BuildingTile.vue:
```diff
@@ -0,0 +1,245 @@
+        </div>
+
+        <div class="text">
+          <g-link :to="path">
```
Yeah, it's a bit odd, but if you tab through the page, this link is the tabbable one, since it has clear text. We don't want the whole card to read out as a link:

![Screenshot from 2025-02-14 18-19-04](https://github.com/user-attachments/assets/f810b02e-ebea-4925-9b46-3c46e0d837f4)

vkoves commented:
@wangbertha - I've addressed your feedback! I don't think we should add another header link (there's so many already) and I think it's fine for that page to come off of a "See more" on the homepage, since that's where you see that data in the first place. We can always tweak it!

I also added the map link under the search that I had in the mocks, but totally forgot about 😆

wangbertha reviewed (approved):
Looks great!

One minor comment related to the article date - I think it would be helpful to add the date to the article itself as well, and as a subtitle in the Blog page. But that can always be a separate update!

vkoves commented:
@wangbertha - great point, I actually added that in with Colton's PR #147, since that ends our second ever blog post!
